packages/labs/ssr/src/lib/server-template.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  html as baseHtml,
+  type CompiledTemplateResult,
+  type TemplateResult,
+} from 'lit-html';
+
+/**
+ * A lit-html template that can only be rendered on the server, and cannot be
+ * hydrated.
+ *
+ * This is a performance optimization, causing the server not to emit comment
+ * markers for updating the dynamic parts of the server-rendered DOM, and
+ * supports rendering into raw text elements like <title> and <textarea>.
+ */
+const SERVER_ONLY = 1;
+
+export interface ServerRenderedTemplate extends TemplateResult {
+  $_litServerRenderMode: typeof SERVER_ONLY;
+}
+
+/**
+ * Returns a variant of the given TemplateResult that, when rendered
+ * server-side, will not be rendered for client-side hydration.
+ *
+ * This is a performance optimization, causing the server not to emit comment
+ * markers for updating the dynamic parts of the server-rendered DOM.
+ * This results in fewer DOM nodes on the client. In most cases the difference
+ * will be small.
+ *
+ * It also allows for data binding into raw text elements like <title> and
+ * <textarea>.
+ *
+ * Currently, a server-only template cannot contain a normal Lit template,
```

Comment:
Is this statement at odds with the description in the README?

> serverhtml template won't send down marker comments, so they can't be updated on the client, however if you render a normal Lit template inside a serverhtml template, then it can be hydrated and updated. Likewise, if you render a custom element inside a serverhtml template, it will automatically hydrate and update like normal.

Reply:
Good catch, yes, that was an out of date comment from an earlier commit, before I added support for normal templates inside server-only templates. Rewrote the comment.